ocaml/lambda/lambda.ml

```diff
@@ -1946,16 +1946,28 @@ let primitive_result_layout (p : primitive) =
   | Pstring_load_128 { boxed = false; _ } | Pbytes_load_128 { boxed = false; _ }
   | Pbigstring_load_128 { boxed = false; _ } ->
       layout_unboxed_vector Pvec128
-  | Pfloatarray_load_128 _ | Pfloat_array_load_128 _
-  | Punboxed_float_array_load_128 _ -> layout_boxed_vector Pvec128
-  | Punboxed_float32_array_load_128 _ -> layout_boxed_vector Pvec128
-  | Pint_array_load_128 _ | Punboxed_int64_array_load_128 _
-  | Punboxed_nativeint_array_load_128 _ ->
-    (* 128-bit types are only supported in the x86_64 backend, so we may
-       assume that nativeint is 64 bits. *)
-    layout_boxed_vector Pvec128
-  | Punboxed_int32_array_load_128 _ ->
-    layout_boxed_vector Pvec128
+  | Pfloatarray_load_128 { boxed = true; _ }
+  | Pfloat_array_load_128 { boxed = true; _ }
+  | Punboxed_float_array_load_128 { boxed = true; _ }
+  | Punboxed_float32_array_load_128 { boxed = true; _ }
+  | Pint_array_load_128 { boxed = true; _ }
+  | Punboxed_int64_array_load_128 { boxed = true; _ }
+  (* 128-bit types are only supported in the x86_64 backend, so we may
```

Comment:
I just noticed this existing comment should be updated; we'll support these for arm64 too.

Reply:
I just deleted the comment, since it's not directly relevant here and the condition is checked at `lambda_to_flambda_primitives:1807`